### PR TITLE
feat: add logger setup

### PR DESCRIPTION
**Pull request by edouardparis**
__env var:__
`REVAULTGUI_DEBUG` (bool)  is used to specify logger level 
and will be used to enable hecrj/iced debug mode.

__logger:__
Only two log level: debug or default.
example: `REVAULTGUI_DEBUG=true ./revault-gui > log.txt`

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -15,6 +36,31 @@ fn main() {
         }
     };
 
+    let debug = match std::env::var("REVAULTGUI_DEBUG") {
+        Ok(var) => match FromStr::from_str(&var) {
+            Ok(v) => v,
+            Err(_) => {
+                println!("Error: REVAULtGUI_DEBUG must be `false` or `true`");
```
Comment (darosior) [inline src/main.rs:43]: nit: REVAULTGUI_DEBUG

File: src/main.rs [hunk 2]
```diff
@@ -15,6 +36,31 @@ fn main() {
...
+                std::process::exit(1);
+            }
+        },
+        Err(VarError::NotUnicode(_)) => {
+            println!("Error: REVAULtGUI_DEBUG must be `false` or `true`");
```
Comment (darosior) [inline src/main.rs:48]: Same

File: src/main.rs [hunk 3]
```diff
@@ -15,6 +36,31 @@ fn main() {
         }
     };
 
+    let debug = match std::env::var("REVAULTGUI_DEBUG") {
+        Ok(var) => match FromStr::from_str(&var) {
+            Ok(v) => v,
```
Comment (darosior) [inline src/main.rs:41]: Just to be sure, so Rust here knows that you want a bool based on the last Err() branch below ?

**Review by darosior — APPROVED** [review]
ACK 5671f5ba7dbf806470a3379724e4c9289d1deaf9 -- Fine as is, but you probably want to:
1. Log as `trace` when sending and receiving stuff to the RPC interface
2. Have two distinct env vars `REVAULTGUI_LOGLEVEL` and `REVAULTGUI_LOGLEVEL`, so that the former can be set to the accurate level (debug, trace, info, warn, ..) and the latter could just imply "at least debug loglevel".

File: src/main.rs (hunk 3 above)
Comment (edouardparis) [reply src/main.rs:41]: Yes, with the  branch `Err(VarError::NotPresent) => false,`

**Review by darosior — APPROVED** [review]
ACK 640e686f5382b1f7c34b5acb068806ceff6cfd98

**Pull request merged**